veps/sig-storage/incremental-backup.md

```diff
@@ -0,0 +1,330 @@
+spec:
+    source:
+        apiGroup: kubevirt.io
+        kind: VirtualMachine
+        name: my-vm
+    incremental: backup1-16210527022025
+status:
+    checkpointName: backup1-15290501032025
+    pvcName: incbackup-my-vm-16210527022025
+```
+
+### Checkpoints
+
+As mentioned two approaches are considered:
+
+- In case CR will be chosen it will look like:
+
+_Checkpoint_
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineCheckpoint
+metadata:
+    name: my-vm-backup-16210527022025
+    namespace: ns1
+spec:
+    xmlData: |
+        <?xml version="1.0" encoding="UTF-8"?>
+        <domaincheckpoint>
+            <name>1525889631</name>
+            <parent>
+                <name>1525111885</name>
+            </parent>
+            <creationTime>1525889631</creationTime>
+            <disks>
+                <disk name='vda' checkpoint='bitmap' bitmap='1525889631'/>
+                <disk name='vdb' checkpoint='no'/>
+            </disks>
+        </domaincheckpoint>
+    OR
+    xmlData: PGRvbWFpbmNoZWNrcG9pbnQ+CiAgICA8bmFtZT4xNTI1ODg5NjMxPC9uYW1lPgogICAgPHBhcmVudD4KICAgICAgICA8bmFtZT4xNTI1MTExODg1PC9uYW1lPgogICAgPC9wYXJlbnQ+CiAgICA8Y3JlYXRpb25UaW1lPjE1MjU4ODk2MzE8L2NyZWF0aW9uVGltZT4KICAgIDxkaXNrcz4KICAgICAgICA8ZGlzayBuYW1lPSd2ZGEnIGNoZWNrcG9pbnQ9J2JpdG1hcCcgYml0bWFwPScxNTI1ODg5NjMxJy8+CiAgICAgICAgPGRpc2sgbmFtZT0ndmRiJyBjaGVja3BvaW50PSdubycvPgogICAgPC9kaXNrcz4KPC9kb21haW5jaGVja3BvaW50Pg== 
+status:
+    creationTime: "2025-03-03T16:13:28Z"
+```
+
+Users can list and retrieve checkpoint information via kubectl.
+In the event of a VM restart, during initialization, the virt-launcher will list the checkpoints CRs and use the XML data to redefine the checkpoints in Libvirt.
+
+- If we choose not to create a CR and instead store the XML in the VM state PVC, we will create a directory specifically for the checkpoint XML files. Each xml file is very minimal and doesn't take much space.
+In the event of a VM restart, we will use the saved checkpoints xml at the time of the restart to redefine them in libvirt.
+To retrieve information about the checkpoints, user can exec virsh command on the virt-launcher pod, or we can use libvirt commands like `virDomainListAllCheckpoints` and `virDomainCheckpointGetXMLDesc` and implement an API to provide this information to the user.
+Alternatively, or additionally, the backup vendor can use the backup's checkpoint name, which will be updated in the VirtualMachineBackup CR status, to manage the checkpoints list themselves, providing their users with checkpoints information through their API.
+
+
+### Collection of the backup
+
+> Note: All of the information mentioned below are suggestions for backup vendors or users to fetch the backup. **There will not be an API provided for this.**
+
+**Push mode:**
+PVC name containing the backup is provided in the VirtualMachineBackup CR status. A data-mover pod can be spawned and copy the data over to a remote storage. After the data is moved, the PVC can be deleted.
```

Comment:
The files will be stored as sparse qcow2 images.  We will document some examples of how to stitch these files together to construct a restorable image.